Source/Android/jni/MainAndroid.cpp

```diff
@@ -293,6 +285,13 @@ Java_org_dolphinemu_dolphinemu_NativeLibrary_IsRunningAndUnpaused(JNIEnv*, jclas
   return static_cast<jboolean>(Core::GetState(Core::System::GetInstance()) == Core::State::Running);
 }
 
+JNIEXPORT jboolean JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_IsUninitialized(JNIEnv*,
+                                                                                        jclass)
+{
+  return static_cast<jboolean>(Core::IsUninitialized(Core::System::GetInstance()) &&
+                               !s_is_booting.IsSet());
+}
+
```

Comment:
Yes, exactly.

Android doesn't want us to do any operations on the main thread that could cause the GUI to drop a frame, so instead of calling `BootManager::BootCore` on the main thread, we launch a separate thread that calls it. But we want the state reported to the main thread to be `State::Starting` as soon as the separate thread has been launched, which is why `s_is_booting` exists.